ChangeLog.md

```diff
@@ -277,6 +277,8 @@ Bug fixes:
   [#4292](https://github.com/commercialhaskell/stack/issues/4292)
 * Fix for git packages to update submodules to the correct state. See
   [#4314](https://github.com/commercialhaskell/stack/pull/4314)
+* Fix to allow dependencies on specific versions of local git repositories. See
```

Comment:
This should go in "unreleased changes" instead. It should also likely be mentioned in the `pantry` changelog.